model/map/src/main/java/org/keycloak/models/map/group/MapGroupProvider.java

```diff
@@ -186,6 +187,18 @@ public Stream<GroupModel> searchForGroupByNameStream(RealmModel realm, String se
             }).sorted(GroupModel.COMPARE_BY_NAME).distinct();
     }
 
+    @Override
+    public Stream<GroupModel> searchGroupsByAttributes(RealmModel realm, Map<String, String> attributes, Integer firstResult, Integer maxResults) {
+        DefaultModelCriteria<GroupModel> mcb = criteria();
+        mcb =  mcb.compare(GroupModel.SearchableFields.REALM_ID, Operator.EQ, realm.getId());
+        for (Map.Entry<String, String> entry : attributes.entrySet()) {
+            mcb = mcb.compare(GroupModel.SearchableFields.ATTRIBUTE, Operator.EQ, entry.getKey(), entry.getValue());
+        }
+
+        return tx.read(withCriteria(mcb).pagination(firstResult, maxResults, SearchableFields.ID ))
```

Comment:
Why are the results ordered by `SearchableFields.ID`? Shouldn't it be rather ordered by `SearchableFields.NAME` ?

Reply:
changed to name